pkg/ccl/changefeedccl/changefeed_dist.go

```diff
@@ -263,7 +263,7 @@ func startDistChangefeed(
 		checkpoint = progress.Checkpoint
 	}
 	p, planCtx, err := makePlan(execCtx, jobID, details, description, initialHighWater,
-		trackedSpans, checkpoint, localState.drainingNodes)(ctx, dsp)
+		trackedSpans, checkpoint, nil, localState.drainingNodes)(ctx, dsp)
```

Comment:
We should pass in the span level checkpoints here - I did that in https://github.com/cockroachdb/cockroach/pull/140379/files, so perhaps you could rebase your work on top of mine to pick that up.